rasa/core/policies/policy.py

```diff
@@ -114,17 +121,37 @@ def _create_featurizer(
 
     def __init__(
         self,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
         featurizer: Optional[TrackerFeaturizer] = None,
```

Comment:
💡 I've decided to keep the `TrackerFeaturizer` separate as it's type changes: in the config it's a dictionary with config parameters - here it's an instantiated Python object. Keeping it out of the config also avoids that we persist the featurizer twice (as part of the config and in the `persist` method).

Reply:
👍  I think that also avoids the problem that the config doesn't contain the domain (so if we wanted to use the config to re-instantiate the `TrackerFeaturizer` instead of loading it from disk and passing it here as a `TrackerFeaturizer`, then we'd need to add the domain to the config before persisting and adapt the `TrackerFeaturizers` constructor to handle the domain - but we could do that later when we disentangle :) )

Reply:
Wait, do we even need to pass it here? If we're using the constructor during training, this is None. If we use it during inference, then the featurizer is accessible via the model_storage, no?

Reply:
@ka-bu The policies would use their `load`  method to instantiate themselves (and their featurizers) from disk. The actual constructor should then no longer have to read anything from disk.

Not sure I'm understanding your previous comment though 👀 The `Domain` would be a separate input to `train` or `predict_next_action_probabilities` - fed by a parent graph node.

Reply:
@wochinge In that first comment I was just trying to go through what we would need to do in case we wanted to put everything in that that config (i.e. not the `TrackerFeaturizer` object in the config but extending the config by a dictionary that can be used to instantiate the `TrackerFeaturizer`). And I guess that idea doesn't make much sense at the moment because we don't have anything to configure there now 🤔  . And I think before asking more questions I should have a look at how/where custom TrackerFeaturizers would be instantiated (I was looking no further as `_standard_featurizer` so far), I will do that tomorrow.
(And in the second comment I was just confusing this here with the approach of loading and then passing e.g. the internal `model` to the constructor again - forget that :)) 
Thanks!